feat(Depot User) Add depot_permissions

lomamech opened the pull request:
feat(Depot User): Add table depot_permission
- Improve and Perf componenr bh-multiple-depot-select
- Update User registry set the option to Assigne Depot to User
- Update Modal to Set Management Depots to Patient
- Implement API in server side to Set depot to User

test(Feature Depot For User)  …
- Implement Integration test for Management depot for User
- Initialise data for table depot_permission


test(Integration test E2E for Depot for User)  …
- Add component for bhMultipleDepot Select
- Implement test E2E for set Depots management by User

closes #2171

lomamech committed:
feat(Depot User): Add table depot_permission

lomamech committed:
Remove shorcut.lnk

lomamech committed:
feature(Depot for User)
- Improve and Perf componenr bh-multiple-depot-select
- Update User registry set the option to Assigne Depot to User
- Update Modal to Set Management Depots to Patient
- Implement API in server side to Set depot to User

lomamech committed:
feature(Depot for User)
- Update API for Depot, Use FilterParser, Select Depot by User
- Update component Depot Select add the Param for the reaserch By User_id

lomamech committed:
test(Feature Depot For User)
- Implement Integration test for Management depot for User
- Initialise data for table depot_permission

lomamech committed:
test(Integration test E2E for Depot for User)
- Add component for bhMultipleDepot Select
- Implement test E2E for set Depots management by User

lomamech committed:
Merge with branch Master

jniles commented on client/src/i18n/fr/depot.json:
```diff
@@ -23,8 +23,8 @@
       "SELECT_A_DEPOT" : "Selectionner un Dépôt",
-      "NO_DEFAULT_DEPOT" : "Vous devez selectionner un depot.",
-      "SELECTION_EXPLANATION" : "Il semble que vous n'avez pas un depot selectionner.  S'il vous plait, selectionnez votre depot dans la liste suivante.  Vous devez le faire une fois.  Si vous ne le trouvez pas, vous pouvez ajouter-le avec ce lien:"
+      "NO_DEFAULT_DEPOT" : "Vous devez selectionner un dépôt.",
```
:+1:  This change makes the application look more professional.  Well done!

jniles commented on client/src/modules/users/templates/grid/action.cell.html:
```diff
@@ -15,6 +15,12 @@
     </li>
+    <li>
+      <a data-method="depot" ng-click="grid.appScope.depotManagement(row.entity)" href>
+        <i class="fa fa-cubes"></i> <span translate>FORM.BUTTONS.DEPOSIT_MANAGEMENT</span>
```
I think this should be "DEPOT_MANAGEMENT", shouldn't it?

jniles commented on server/controllers/inventory/depots/index.js:
```diff
@@ -110,12 +111,27 @@ function update(req, res, next) {
-    WHERE enterprise_id = ?;`;
+  `;
+
+  filters.custom(
```
That's interesting!  I never thought about adding this as a feature on the depots controller.

jniles commented on server/controllers/inventory/depots/index.js:
```diff
@@ -110,12 +111,27 @@ function update(req, res, next) {
+  const options = req.query;
+  options.enterprise_id = req.session.enterprise.id;
+
+  const filters = new FilterParser(options, { tableAlias : 'depot' });
```
:+1:

jniles commented on client/src/modules/stock/depot-selection.ctrl.js:
```diff
@@ -26,8 +26,10 @@ function SelectDepotModalController(Instance, Depots, Notify, depot) {
   function startup() {
     toggleLoadingIndicator();
+    // download only the depots that the user has the management right
+    var userId = Session.user.id;
```
:+1:

jniles reviewed (commented):
@lomamech, this is really good!

The only comment I have - could you add a message on the depot selection modal that says "These are the depots that you have permission for"?  That way, a user doesn't get confused why not all depots are listed.

Otherwise, it all looks good.  Great work!

lomamech committed:
refactor(User Registry)
- change translate key Use DEPOT_MANAGEMENT instead DEPOSIT MANAGEMENT
- Update the message on the depot selection Modal

jniles reviewed (approved):
LGTM.
